src/pathreduction.rs

```diff
@@ -48,22 +48,19 @@ pub fn reduce_paths(paths: &[PathBuf]) -> HashSet<PathBuf> {
         })
         .filter(|reduction| reduction != &PathReduction::Remove)
         .map(|reduction| reduction.unwrap("previous filter got them"))
-        .collect::<HashSet<PathBuf>>()
-        .into_iter()
         .collect::<Vec<PathBuf>>();
 
     // Sort by length so we automatically select project roots when
     // possible, in the next fold.
     reduced.sort();
+    reduced.dedup();
     reduced
         .into_iter()
         .fold::<HashSet<PathBuf>, _>(HashSet::new(), |mut set, new_path| {
-            if set.iter().any(|path| new_path.starts_with(path)) {
-                set
-            } else {
+            if !set.iter().any(|path| new_path.starts_with(path)) {
```

Comment:
```suggestion
            if !set.iter().any(new_path.starts_with) {
```

Reply:
I haven't tested your suggestion, but I believe it wouldn't compile. You can eliminate the closure when there's no self, but there is here. Rust would have to implicitly create a closure that passes new_path along (but it doesn't).

Reply:
You are right, can’t call methods that way.